modules/api/php/endpoints/candidate/visit/dicoms.class.inc

```diff
@@ -153,6 +171,170 @@ class Dicoms extends Endpoint implements \LORIS\Middleware\ETagCalculator
+            );
+        }
+
+        $pscid = $data['PSCID'] ?? null;
+        if (!\Candidate::candidateExists($visit_candid, $pscid)) {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                'PSCID and CandID in POST data do not refer to the same candidate.'
+            );
+        }
+
+        $visit_label = $data['VisitLabel'] ?? null;
+        if ($visit_label !== $this->_visit->getVisitLabel()) {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                'Visit label in URL and POST data do not match.'
+            );
+        }
+
+        $is_phantom = $data['IsPhantom'] ?? null;
+        if (!is_bool($is_phantom)) {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                'IsPhantom must be a boolean.'
+            );
+        }
+
+        $overwrite = $data['Overwrite'] ?? false;
+        if (!is_bool($overwrite)) {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                'Overwrite must be a boolean.'
+            );
+        }
+
+        $file_name    = $uploaded_file->getClientFilename();
+        $patient_name = $pscid . "_" . $input_candid . "_" . $visit_label;
+        if (!preg_match("/^{$patient_name}/", $file_name) && !$is_phantom) {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                "Filename must start with $patient_name."
+            );
+        }
+
+        // *************************
+        // *** Upload file *********
+        // *************************
+        $incoming_path = \NDB_factory::singleton()
+            ->config()
+            ->getSetting('MRIUploadIncomingPath');
+
+        $incoming_dir = new \SplFileInfo($incoming_path);
+
+        $uploader = (new \LORIS\FilesUploadHandler($incoming_dir))
+            ->withPermittedMIMETypes(
+                'application/gzip',
+                'application/tar',
+                'application/tar+gzip',
+                'application/x-tar'
+            );
+
+        if ($overwrite) {
+            $uploader = $uploader->withOverwrite(true);
+        }
+
+        $upload_response = $uploader->handle($request);
+        if ($upload_response->getStatusCode() !== 201) {
+            return $upload_response;
+        }
+
+        // *************************
+        // *** Update Database *****
+        // *************************
+        $file_path = $incoming_dir->getPathname() . '/' . $file_name;
+        $file_info = new \SplFileInfo($file_path);
+        if (!$file_info->isFile()) {
+            return new \LORIS\Http\Response\JSON\InternalServerError(
+                'Upload failed, please contact your LORIS administrator.'
+            );
+        }
+
+        try {
+            $this->_visit->addMRIUpload(
+                $user,
+                $file_info,
+                $patient_name,
+                $is_phantom,
+            );
+        } catch (\Throwable $e) {
+            error_log($e->getMessage());
```

Comment:
Should use the logger

Reply:
https://github.com/aces/Loris/pull/9154/commits/0488bb032d190bc9fdb229cf6cd6d3057e418c5a